# Issue 28131: latte interface broken with Python 3

archive/issues_027894.json:
```json
{
    "body": "```\n**********************************************************************\nFile \"latte.py\", line 52, in sage.interfaces.latte.count\nFailed example:\n    count(P.cdd_Hrepresentation(), cdd=True)   # optional - latte_int\nException raised:\n    Traceback (most recent call last):\n      File \"/opt/sage/sage-py3-gcc/local/lib/python3.7/site-packages/sage/doctest/forker.py\", line 681, in _run\n        self.compile_and_execute(example, compiler, test.globs)\n      File \"/opt/sage/sage-py3-gcc/local/lib/python3.7/site-packages/sage/doctest/forker.py\", line 1105, in compile_and_execute\n        exec(compiled, globs)\n      File \"<doctest sage.interfaces.latte.count[2]>\", line 1, in <module>\n        count(P.cdd_Hrepresentation(), cdd=True)   # optional - latte_int\n      File \"/opt/sage/sage-py3-gcc/local/lib/python3.7/site-packages/sage/interfaces/latte.py\", line 153, in count\n        ans, err = latte_proc.communicate(arg)\n      File \"/opt/sage/sage-py3-gcc/local/lib/python3.7/subprocess.py\", line 939, in communicate\n        stdout, stderr = self._communicate(input, endtime, timeout)\n      File \"/opt/sage/sage-py3-gcc/local/lib/python3.7/subprocess.py\", line 1666, in _communicate\n        input_view = memoryview(self._input)\n    TypeError: memoryview: a bytes-like object is required, not 'str'\n**********************************************************************\n```\n\nCC:  @fchapoton @mkoeppe\n\nKeywords: py3\n\nBranch/Commit: 579e9ef6c4cc66bfcbbc0fc79d5c68153335a561\n\nReviewer: Fr\u00e9d\u00e9ric Chapoton\n\nAuthor: Vincent Delecroix\n\nResolution: fixed\n\nIssue created by migration from https://trac.sagemath.org/ticket/28131\n\n",
    "closed_at": "2019-07-11T18:54:00Z",
    "created_at": "2019-07-08T04:20:58Z",
    "labels": [
        "component: interfaces",
        "bug"
    ],
    "milestone": "https://github.com/sagemath/sagetest/milestones/sage-8.9",
    "title": "latte interface broken with Python 3",
    "type": "issue",
    "url": "https://github.com/sagemath/sagetest/issues/28131",
    "user": "https://github.com/videlec"
}
```
```
**********************************************************************
File "latte.py", line 52, in sage.interfaces.latte.count
Failed example:
    count(P.cdd_Hrepresentation(), cdd=True)   # optional - latte_int
Exception raised:
    Traceback (most recent call last):
      File "/opt/sage/sage-py3-gcc/local/lib/python3.7/site-packages/sage/doctest/forker.py", line 681, in _run
        self.compile_and_execute(example, compiler, test.globs)
      File "/opt/sage/sage-py3-gcc/local/lib/python3.7/site-packages/sage/doctest/forker.py", line 1105, in compile_and_execute
        exec(compiled, globs)
      File "<doctest sage.interfaces.latte.count[2]>", line 1, in <module>
        count(P.cdd_Hrepresentation(), cdd=True)   # optional - latte_int
      File "/opt/sage/sage-py3-gcc/local/lib/python3.7/site-packages/sage/interfaces/latte.py", line 153, in count
        ans, err = latte_proc.communicate(arg)
      File "/opt/sage/sage-py3-gcc/local/lib/python3.7/subprocess.py", line 939, in communicate
        stdout, stderr = self._communicate(input, endtime, timeout)
      File "/opt/sage/sage-py3-gcc/local/lib/python3.7/subprocess.py", line 1666, in _communicate
        input_view = memoryview(self._input)
    TypeError: memoryview: a bytes-like object is required, not 'str'
**********************************************************************
```

CC:  @fchapoton @mkoeppe

Keywords: py3

Branch/Commit: 579e9ef6c4cc66bfcbbc0fc79d5c68153335a561

Reviewer: Frédéric Chapoton

Author: Vincent Delecroix

Resolution: fixed

Issue created by migration from https://trac.sagemath.org/ticket/28131





---

archive/issue_comments_393457.json:
```json
{
    "body": "Changing status from new to needs_review.",
    "created_at": "2019-07-08T04:32:33Z",
    "issue": "https://github.com/sagemath/sagetest/issues/28131",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sagetest/issues/28131#issuecomment-393457",
    "user": "https://github.com/videlec"
}
```

Changing status from new to needs_review.



---

archive/issue_comments_393458.json:
```json
{
    "body": "<a id='comment:2'></a>Branch pushed to git repo; I updated commit sha1. Last 10 new commits:",
    "created_at": "2019-07-08T04:32:36Z",
    "issue": "https://github.com/sagemath/sagetest/issues/28131",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sagetest/issues/28131#issuecomment-393458",
    "user": "https://trac.sagemath.org/admin/accounts/users/git"
}
```

<a id='comment:2'></a>Branch pushed to git repo; I updated commit sha1. Last 10 new commits:



---

archive/issue_comments_393459.json:
```json
{
    "body": "<a id='comment:3'></a>Branch pushed to git repo; I updated commit sha1. This was a forced push. New commits:",
    "created_at": "2019-07-08T04:34:21Z",
    "issue": "https://github.com/sagemath/sagetest/issues/28131",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sagetest/issues/28131#issuecomment-393459",
    "user": "https://trac.sagemath.org/admin/accounts/users/git"
}
```

<a id='comment:3'></a>Branch pushed to git repo; I updated commit sha1. This was a forced push. New commits:



---

archive/issue_comments_393460.json:
```json
{
    "body": "<a id='comment:4'></a>ok, makes sense",
    "created_at": "2019-07-09T18:54:21Z",
    "issue": "https://github.com/sagemath/sagetest/issues/28131",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sagetest/issues/28131#issuecomment-393460",
    "user": "https://github.com/fchapoton"
}
```

<a id='comment:4'></a>ok, makes sense



---

archive/issue_comments_393461.json:
```json
{
    "body": "Changing status from needs_review to positive_review.",
    "created_at": "2019-07-09T18:54:21Z",
    "issue": "https://github.com/sagemath/sagetest/issues/28131",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sagetest/issues/28131#issuecomment-393461",
    "user": "https://github.com/fchapoton"
}
```

Changing status from needs_review to positive_review.



---

archive/issue_events_069817.json:
```json
{
    "actor": "https://github.com/vbraun",
    "created_at": "2019-07-11T18:54:00Z",
    "event": "closed",
    "issue": "https://github.com/sagemath/sagetest/issues/28131",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sagetest/issues/28131#event-69817"
}
```



---

archive/issue_comments_393462.json:
```json
{
    "body": "Resolution: fixed",
    "created_at": "2019-07-11T18:54:00Z",
    "issue": "https://github.com/sagemath/sagetest/issues/28131",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sagetest/issues/28131#issuecomment-393462",
    "user": "https://github.com/vbraun"
}
```

Resolution: fixed
